AutoSteamApp/Automaton/SteamworkAutomaton.cs

```diff
@@ -0,0 +1,348 @@
+            // Press the keys
+            Log.Debug("Random Sequence: [" + string.Join(", ", sequence.Select(x => x.ToString())) + "]");
+            // Wait until we have focus
+            if (!_Process.HasFocus())
+                Log.Message("Waiting for MHW to have focus.");
+            while (!_Process.HasFocus() && !cts.IsCancellationRequested) { };
+
+            for (int i = 0; i < sequence.Length; i++)
+            {
+                StaticHelpers.PressKey(_InputSimulator, sequence[i], ConfigurationReader.RandomInputDelay);
+                Thread.Sleep(ConfigurationReader.RandomInputDelay);
+            }
+            StaticHelpers.PressKey(_InputSimulator, VirtualKeyCode.SPACE, ConfigurationReader.RandomInputDelay);
+            Thread.Sleep(ConfigurationReader.RandomInputDelay);
+            StaticHelpers.PressKey(_InputSimulator, VirtualKeyCode.VK_X, ConfigurationReader.RandomInputDelay);
+        }
+
+
+        #endregion
+
+        #region Helpers
+
+        /// <summary>
+        /// Checks if we have met the satisfying conditions provided by the config file.
+        /// </summary>
+        /// <returns></returns>
+        private bool CheckForExitCondition()
+        {
+            if (ConfigurationReader.OnlyUseNaturalFuel)
+            {
+                // true if we have either equal to or less fuel than specified in the config file.
+                if (_SaveData.NaturalFuelLeft <= ConfigurationReader.StopAtFuelAmount)
+                {
+                    Log.Message("Hit minimum natural fuel reserve.");
+                    return true;
+                }
+                return false;
+            }
+            else
+            {
+                // true if we have either equal to or less fuel than specified in the config file.
+                if (_SaveData.StoredFuelLeft <= ConfigurationReader.StopAtFuelAmount)
+                {
+                    Log.Message("Hit minimum stored fuel reserve.");
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Loads and verifies the current MHW:IB process against the currently supported version.
+        /// </summary>
+        /// TODO: Make it more oo by making this the constructor for an abstract automaton.
+        /// Then have inhertied automatons work based on different versions.
+        /// This would allow for backward compatability
+        /// 
+        private void LoadAndVerifyProcess()
+        {
+            Log.Message("Attempting to load MHW:IB Process.");
+            // First set the mhw process
+            _Process = StaticHelpers.GetMHWProcess();
+
+            Log.Message("Process Loaded. Retrieving version");
+            // Now verify the version
+            Match match = Regex.Match(_Process.MainWindowTitle, MHWMemoryValues.SupportedVersionRegex);
+            // If the match is made
+            if (match.Success)
```

Comment:
`If` clauses can be combined. Also, for readability purposes, you should use the brackets.